From cc142f61a277c3d37f065cd343feb3b6a84d5304 Mon Sep 17 00:00:00 2001
From: liqingqing_1229 <liqingqing3@huawei.com>
Date: Fri, 15 Sep 2023 15:30:44 +0800
Subject: [PATCH] elf: Fix _dl_close_worker compile error with -Werror and
 -DNDEBUG

---
 elf/dl-close.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf/dl-close.c b/elf/dl-close.c
index ea62d0e601..afd1c6b000 100644
--- a/elf/dl-close.c
+++ b/elf/dl-close.c
@@ -137,7 +137,7 @@ _dl_close_worker (struct link_map *map, bool force)
   dl_close_state = pending;
 
   bool any_tls = false;
-  const unsigned int nloaded = ns->_ns_nloaded;
+  __attribute__ ((unused)) const unsigned int nloaded = ns->_ns_nloaded;
 
   /* Run over the list and assign indexes to the link maps.  */
   int idx = 0;
-- 
2.33.0

